fieldtracing/fieldtracing.cpp

```diff
@@ -763,392 +760,191 @@ namespace FieldTracing {
-               mpiGrid[id]->parameters[CellParams::CONNECTION] = TracingPointConnectionType::CLOSED_LOOP;
-            }
-            if (reducedCellFWConnection[n] == TracingLineEndType::LOOP && reducedCellBWConnection[n] == TracingLineEndType::CLOSED) {
-               mpiGrid[id]->parameters[CellParams::CONNECTION] = TracingPointConnectionType::LOOP_CLOSED;
-            }
-            if (reducedCellFWConnection[n] == TracingLineEndType::OPEN && reducedCellBWConnection[n] == TracingLineEndType::LOOP) {
-               mpiGrid[id]->parameters[CellParams::CONNECTION] = TracingPointConnectionType::OPEN_LOOP;
-            }
-            if (reducedCellFWConnection[n] == TracingLineEndType::LOOP && reducedCellBWConnection[n] == TracingLineEndType::OPEN) {
-               mpiGrid[id]->parameters[CellParams::CONNECTION] = TracingPointConnectionType::LOOP_OPEN;
-            }
-            if (reducedCellFWConnection[n] == TracingLineEndType::LOOP && reducedCellBWConnection[n] == TracingLineEndType::LOOP) {
-               mpiGrid[id]->parameters[CellParams::CONNECTION] = TracingPointConnectionType::LOOP_LOOP;
-            }
+         
+         // Now, after stepping, if it is no longer in our domain, another MPI rank will pick up later.
+         if(fsgridCell[0] == -1) {
+            cellNeedsContinuedTracing[n] = 1;
+            cellTracingCoordinates[n] = x;
+            break;
          }
-      }
-      phiprof::stop("final-loop");
-      
-      phiprof::stop("fieldtracing-fullTracing");
+      } // while true
    }
    
-   
-   /*! Trace magnetic field lines forward and backward from each DCCRG cell to record the connectivity.
+   /*!< \brief Trace magnetic field lines forward and backward from each DCCRG cell to record the connectivity and detect flux ropes.
+    *
+    * Full box connection and flux rope tracing
+    *
+    * We are doing two things in one function here as we save a *lot* especially in MPI reductions by combining those. And they are
+    * doing very similar things anyway.
+    *
+    * Firstly we are interested in knowing the ultimate magnetic connection forward and backward from any given location. The
+    * locations are the DCCRG cell centers, there's too many fsgrid cells for practical purposes.
+    * Therefore we step along the magnetic field until we hit an outer box wall (OPEN) or the inner boundary
+    * radius (CLOSED). In rare cases the line might not have reached either of these two termination conditions by the time the
+    * field line has reached a length of maxTracingDistance. In that case we call it DANGLING, it likely ended up in a loop
+    * somewhere (we don't call it "loop" to avoid confusion with the flux rop tracing). As long as we have not hit any of the above
+    * termination conditions, the type is called UNPROCESSED. We allow fieldTracingParameters.fullbox_max_incomplete_lines to remain
```

Comment:
So this is a bit confusing. How can we not hit one of OPEN,CLOSED or DANGLING  criteria?

Reply:
This is what 883 sqq. is explaining: with the allowed leftover fraction.